### ISSUE TITLE
Windows [7] run without Cygwin?

### ISSUE THREAD

**Issue by tonypdmtr** [dmsc/emu2#19]
This is exactly the program I've looking for (for years) as I have many DOS-era utilities that I need to run under Windows 7 console.

[I haven't been able to make DosBox[-x] work with text-only apps in a simple manner.]

I managed to get this emulator work straight out of the box by compiling under Gycwin, and running it there.
But, I want to be able to run it under Windows console directly.  It fails.  I suppose it needs to be compiled differently.

Any hints on that?

**Comment by dmsc** [comment]
Hi @tonypdmtr !

I wrote the code for an Unix OS, it does a few assumptions:
- It uses terminal-control codes for writing to the screen - this is only supported on latest Windows 10 consoles.
- Uses "fork" for running multiple instances of the emulator.
- Has logic to allow case sensitivity on filenames - on Windows filenames are case insensitive.

I won't oppose to a native Windows port, and I can help anybody interested in porting, but I won't write the port myself.

Have Fun!

**Comment by strikersix23** [comment]
I got it to compile and run the issue i had was with the cmd prompt was having scattered rendering just like with qemu no graphic im looking for a fix. Compilation was done using msys2

**Comment by johnsonjh** [comment]
I looked into a port, but the most complex will be to replace the use of signals with NT native timers.  The git source code has an implementation of setitimer and such that could be borrowed.

**Comment by dmsc** [comment]
Yes, the use of signals simplify a lot of the code, as it allows the emulator to use blocking I/O and release the CPU; but also be able to emulate DOS interrupts.

*many* DOS programs depend on interrupts for reading the keyboard and redrawing the screen!

Seeing the git for windows code, it is not complicated, it simply spawns a thread that periodically calls `mingw_raise(SIGALRM)`

Have Fun!

**Comment by strikersix23** [comment]
I got it working with minor issues awhile back it's been forever since then
